browser/ui/views/speedreader/speedreader_bubble_global.cc

```diff
@@ -0,0 +1,158 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/views/speedreader/speedreader_bubble_global.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "base/bind.h"
+#include "base/notreached.h"
+#include "base/strings/utf_string_conversions.h"
+#include "brave/browser/ui/speedreader/speedreader_bubble_controller.h"
+#include "brave/browser/ui/views/speedreader/speedreader_bubble_util.h"
+#include "brave/common/url_constants.h"
+#include "brave/grit/brave_generated_resources.h"
+#include "chrome/browser/ui/views/location_bar/location_bar_bubble_delegate_view.h"
+#include "chrome/grit/generated_resources.h"
+#include "components/strings/grit/components_strings.h"
+#include "content/public/browser/page_navigator.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/browser/web_contents_user_data.h"
+#include "content/public/browser/web_ui_controller.h"
+#include "content/public/common/referrer.h"
+#include "include/core/SkColor.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/base/window_open_disposition.h"
+#include "ui/events/event.h"
+#include "ui/views/controls/button/md_text_button.h"
+#include "ui/views/controls/button/toggle_button.h"
+#include "ui/views/controls/label.h"
+#include "ui/views/controls/link.h"
+#include "ui/views/controls/styled_label.h"
+#include "ui/views/layout/box_layout.h"
+#include "ui/views/metadata/metadata_impl_macros.h"
+#include "ui/views/view.h"
+
+namespace {
+
+constexpr int kBubbleWidth = 324;  // width is 324 pixels
+
+constexpr int kFontSizeSiteTitle = 14;  // site title font size
+
+}  // anonymous namespace
+
+namespace speedreader {
+
+SpeedreaderBubbleGlobal::SpeedreaderBubbleGlobal(
+    views::View* anchor_view,
+    content::WebContents* web_contents,
+    SpeedreaderBubbleController* controller)
+    : SpeedreaderBubbleView(anchor_view),
+      web_contents_(web_contents),
+      controller_(controller) {
+  SetButtons(ui::DialogButton::DIALOG_BUTTON_NONE);
+}
+
+void SpeedreaderBubbleGlobal::Show() {
+  ShowForReason(USER_GESTURE);
+}
+
+void SpeedreaderBubbleGlobal::Hide() {
```

Comment:
Bubble will be hidden and destroyed when it's de-activated.
So, I think we don't need to call `Hide` explicitely?

Reply:
I read `QRCodeGeneratorBubbleController` and it explicitely hide from its dtor.
It makes sense to call Hide() from there.